flake.nix

```diff
@@ -200,23 +199,9 @@
                             cmds = [
                               ''echo "Deploying flake to: {{.HOSTNAME}}"''
                               ''
-                                source_path={{.LOCAL_FLAKE_SOURCE}}
-                                if test -d "$source_path"; then
-                                  source_path=$source_path/
-                                fi
-                                ${pkgs.rsync}/bin/rsync \
-                                --verbose \
-                                -e "{{.REMOTE_COMMAND}} -l {{.REMOTE_USER}} -T {{.REMOTE_SSH_OPTS}}" \
-                                -FD \
-                                --checksum \
-                                --times \
-                                --perms \
-                                --recursive \
-                                --links \
-                                --delete-excluded \
-                                --mkpath \
-                                ${optionalString useSudo ''--rsync-path="{{.REMOTE_SUDO_COMMAND}} {{.REMOTE_SUDO_OPTS}} rsync"''} \
-                                $source_path {{.REMOTE_USER}}\@{{.REMOTE_HOST}}:{{.REMOTE_CONFIG_DIR}}
+                                NIX_SSHOPTS="{{.REMOTE_SSH_OPTS}}" nix flake archive \
+                                  --to ssh://{{.REMOTE_USER}}@{{.REMOTE_HOST}} \
+                                  {{.LOCAL_FLAKE_SOURCE}}
```

Comment:
I would argue that the deployment action should create a symlink that is easy for the user to find to the deployed flake.
We could re-use config-dir for that, e.g. be `/var/lib/lollypops` be a symlink to the /nix/store path containing `{{.LOCAL_FLAKE_SOURCE}}`

Apart from being a bit cleaner in the `rebuild` action (see comment above), It would preserve the abilitiy for users to rebuild on the remote itself, which has proved useful for me in the past after deploying a configuration that broken network stuff for example.